server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -236,7 +236,7 @@ public ShardSearchRequest(StreamInput in) throws IOException {
         shardRequestIndex = in.getVersion().onOrAfter(Version.V_7_11_0) ? in.readVInt() : -1;
         numberOfShards = in.readVInt();
         scroll = in.readOptionalWriteable(Scroll::new);
-        source = in.readOptionalWriteable(SearchSourceBuilder::new);
+        source(in.readOptionalWriteable(SearchSourceBuilder::new));
```

Comment:
Sorry I wasn't clear, I was referring to the other constructors above that takes the source as an argument. We should always create the request with the minimal source.